howso/hierarchy.amlg

```diff
@@ -811,36 +811,78 @@
 		(call !Return (assoc payload (assoc "count" (size cases_to_move)) ))
 	)
 
-	;method to update the references for contained trainees, should only be used for debugging purposes.
-	;{idempotent (true)}
-	#set_contained_trainee_maps
+	;method to update the references for contained trainees
+	;{idempotent (true) protected (true)}
+	#set_hierarchy_relationships
 	(declare
 		(assoc
+			;{type ["string" "null"]}
+			;the unique string identifier for the parent of the trainee
+			parent_id (null)
 			;{type "assoc" additional_indices "string"}
```

Comment:
Might also type these maps for "null" too, since it looks like they work the same as parent_id